### ISSUE TITLE
[AutoResearch] on-device emotion inference — full privacy, zero server dependency

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#233]
## What
Move all emotion inference (voice + health) to run entirely on-device so user data never leaves the phone.

## Why
Privacy is the #1 concern for mental health apps. Replika was fined €5.6M by Italy's privacy watchdog (2025). On-device inference = no server, no data upload, no privacy risk. This is also a competitive moat — Hume AI and most competitors require cloud processing.

We already have ONNX emotion model running locally (2.2MB). Need to extend this to voice emotion + health analysis.

## Research Findings
- Edge AI + federated learning: 200+ papers since 2021, convergence of privacy + low latency (Preprints.org 2025)
- Pattern: large transformer as feature extractor + simple linear classifier = competitive accuracy with real-time on-device inference (PMC 2021)
- 2025 trend: on-device AI is the "decentralized intelligence revolution" — privacy + speed + offline
- Differential privacy adds noise during processing, preventing inference of sensitive data
- Apple's approach: on-device ML for Health features, no data leaves iPhone

### What Can Run On-Device Today
| Component | On-Device? | How |
|---|---|---|
| Voice emotion (emotion2vec+) | Partial — needs ONNX export | Export emotion2vec to ONNX, run in JS/WASM |
| Voice biomarkers (jitter/shimmer/HNR) | Yes | Pure DSP, no ML needed |
| Health data analysis | Yes | Simple math on Apple Health data |
| EI composite scoring | Yes | Weighted averages, no ML |
| Supplement correlation | Yes | Statistical computation |
| Sleep-mood prediction | Yes — after training | Small model, export to ONNX |
| EEG emotion (mega LGBM) | Already done | ONNX in client/public/models/ |

### What Needs Work
1. **Export emotion2vec+ to ONNX** — the voice model currently requires PyTorch server-side
   - Alternative: use DistilHuBERT (23MB, already supported) as on-device fallback
   - Alternative: extract MFCC/biomarkers in JS, run small ONNX classifier
2. **WASM inference runtime** — use onnxruntime-web for browser/Capacitor
3. **On-device feature extraction** — implement Welch PSD, band power extraction in JavaScript
4. **Federated learning for personalization** — train personal model on-device, only share gradients

### Architecture
```
Phone (all processing local)
├── Microphone → voice features → ONNX classifier → emotion
├── Apple Health → HRV/sleep/activity → statistical analysis → stress/recovery
├── Supplement log → local correlation engine → verdicts
├── Optional EEG → ONNX LGBM → emotion (already works)
└── All results → local EI composite → Daily Brain Report

Server (optional, only for):
├── AI emotion coach (LLM needs cloud)
├── Federated model updates (periodic, anonymized gradients only)
└── Data backup (encrypted, opt-in)
```

## Files
- Modify: `client/src/hooks/use-inference.ts` — extend local-first pattern to voice
- Create: `client/src/lib/voice-inference.ts` — on-device voice feature extraction
- Create: `client/src/lib/health-inference.ts` — on-device health analysis
- Export: emotion2vec ONNX model or use DistilHuBERT ONNX
- Modify: multimodal EI — client-side computation option

## Sources
- [Edge AI Survey 2021-2025 (Preprints.org)](https://www.preprints.org/manuscript/202512.0118)
- [On-Device Emotion Recognition (PMC 2021)](https://pmc.ncbi.nlm.nih.gov/articles/PMC8271649/)
- [On-Device AI Models Survey (ACM 2025)](https://dl.acm.org/doi/10.1145/3724420)
- [Federated Learning for Privacy-Preserving Edge AI (ResearchGate 2025)](https://www.researchgate.net/publication/392040839_Federated_Learning_for_Privacy-Preserving_Edge_AI)

## Complexity
Large — ONNX export + JS inference + architecture shift

**Comment by LakshmiSravyaVedantham** [comment]
Implemented: `ml/api/routes/on_device.py` is registered and live on main. Closing.
